server/service/ms/datasource.go

```diff
@@ -18,10 +18,11 @@ package ms
 import (
 	"context"
 	pb "github.com/apache/servicecomb-service-center/pkg/registry"
+	"github.com/apache/servicecomb-service-center/server/plugin/registry"
 )
 
 type DataSource interface {
-	RegisterService(ctx context.Context, service *pb.CreateServiceRequest) (*pb.CreateServiceResponse, error)
+	RegisterService(ctx context.Context, service *pb.CreateServiceRequest) (*registry.PluginResponse, error)
```

Comment:
why? PluginResponse is a kv spec ,why not use bizz model

Reply:
Because I think that, the result(*PluginResponse) of dao layer operation need to be checked and returned in datasource  module, if the CreateServiceResponse wants to be returned here. 
However, the datasource module is designed to act like shim between dao layer and api bizz layer, **providing sc with ability for developer to design different databse kv format according the advantanges of databases**. So this module shuoldn't take part in the bussiness operation.

Reply:
decouple etcd model later please

Reply:
ok,I will decouple it and migrate registry next commit